content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: blog
+title: "Node Log Viewer"
+date: 2023-06-03
+slug: node-log-viewer-alpha
+---
+
+**Author:** Aravindh Puthiyaparambil (Red Hat)
+
+Kubernetes 1.27 introduced a new feature called _Node log query_ that allows
+viewing logs of services running on the node.
+
+## What problem does it solve?
+Cluster administrators face issues when debugging malfunctioning services
+running on the node. They usually have to SSH or RDP into the node to view the
+logs of the service to debug the issue. The _Node log query_ feature helps with
+this scenario by allowing the cluster administrator to view the logs using
+_kubectl_. This is especially useful with Windows nodes where you run into the
+issue of the node going to the ready state but containers not coming up due to
+CNI misconfigurations and other issues that are not easily identifiable by
+looking at the Pod status.
+
+## How does it work?
+
+The kubelet already has a _/var/log/_ viewer that is accessible via the node
+proxy endpoint. The feature supplements this endpoint with a shim that shells
+out to _journalctl_, on Linux nodes, and _Get-WinEvent_ cmdlet on Windows
+nodes. It then uses the existing filters provided by the commands to allow
+filtering the logs. The kubelet also uses heuristics to retrieve the logs.
+If the user is not aware if a given system services logs to a file or to the
+native system logger, the heuristics first checks the native operating system
+logger and if that is not available it attempts to retrieve the first logs
+from `/var/log/<servicename>` or `/var/log/<servicename>.log` or
+`/var/log/<servicename>/<servicename>.log`.
+
+On Linux we assume that service logs are available via journald. On Windows we
+assume that service logs are available in the application log provider. Note
+that this feature requires you to have access to node objects.
```

Comment:
```suggestion
On Linux we assume that service logs are available via journald, and that
`journalctl` is installed. On Windows we assume that service logs are available in the
application log provider. Also note that fetching node logs is only available if you are authorized
to do so (in RBAC, that's **get** and **create** access to `nodes/proxy`). The privileges that you
need to fetch node logs also allow elevation-of-privilege attacks, so be careful about how you 
manage them.
```

Reply:
I'd agree that cluster operators will need to be very careful if the permission required is create and get on node/proxy. Those rights allow users to directly contact the kubelet API (bypassing auditing and admission control) and execute commands in any pod on a node (amongst other things)

Reply:
I want to call out that the ability read logs using the `/proxy/logs` endpoint is already available and enabled by default. This feature is extending it be able to read journald logs and defaults to false.